pkg/oc/bootstrap/docker/openshift/helper.go

```diff
@@ -976,3 +992,33 @@ kubelet --help | grep -- "--cgroup-driver"
 
 	return rc == 0 && err == nil
 }
+func (h *Helper) configureHostDockerTrust(registryURL string) error {
+	glog.V(1).Info("Copy master CA to host docker certificates location")
+	certPath := GetRegistryCertPath(registryURL)
+	if _, err := os.Stat(filepath.Dir(certPath)); os.IsNotExist(err) {
+		err := os.Mkdir(filepath.Dir(certPath), hostDockerCertPermissions)
+		if err != nil {
+			return err
+		}
+	}
+
+	id, err := h.runHelper.New().Image(h.image).
+		DiscardContainer().
+		HostPid().
+		Privileged().
+		Bind(registryTrustContainersBinds...).
+		Entrypoint("cp").
+		Command(serverCAPath, certPath).Run()
+	if err != nil || id != 0 {
+		return errors.NewError("cannot copy registry ca to host docker trust folder").WithCause(err)
+	}
+	return err
+}
+
+// GetRegistryCertPath return where we should put registry certificate as per current runtime version based on reg url (/etc/docker/certs.d/[reg_url]/ca.crt)
+func GetRegistryCertPath(registryURL string) string {
+	if _, err := os.Stat(hostDockerCertPath); err == nil {
+		return fmt.Sprintf("%s/%s/%s", hostDockerCertPath, registryURL, HostDockerCertName)
+	}
+	return fmt.Sprintf("%s/%s/%s", hostDockerLatestCertPath, registryURL, HostDockerCertName)
```

Comment:
You can't perform this check on the client because you may not have access to the host's file system. Also hard-coding docker-latest seems fragile, though I don't know if we have a better alternative.

Reply:
without checks, we will be assuming that copying went fine and we print message. I think we can ignore this and just print it.

Reply:
You can perform the check on the server through the container. You could stream a shell script to the container to figure out the cert path, create the directory if needed and copy the file. 

An example is here:
https://github.com/openshift/origin/blob/master/pkg/oc/bootstrap/docker/openshift/cnetwork.go